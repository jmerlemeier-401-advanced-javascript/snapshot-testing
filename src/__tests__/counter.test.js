import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Renderer from 'react-test-renderer';
import Counter from '../components/counter/counter';


Enzyme.configure({ adapter: new Adapter() });

describe('Testing our Counter Component - has State of 0 with Load.', () => {
  it ('Counter state is initially 0', () => {
    let app = mount(<Counter />);
    let button = app.find('button');
    expect(app.state('count')).toBe(0);
  });

  it('Simulate a button click - has State of 1.', () => {
    let app = mount(<Counter />); //this mount is a like a 'refresh' it renders it fresh.
    let button = app.find('.up');
    button.simulate('click');
    expect(app.state('count')).toBe(1);
  });

  //Snapshot data
  it('Renders correctly', () => {
    const renderTree = Renderer.create(<Counter />).toJSON(); //snapshot library created rendertree
    expect(renderTree).toMatchSnapshot();
  })

});