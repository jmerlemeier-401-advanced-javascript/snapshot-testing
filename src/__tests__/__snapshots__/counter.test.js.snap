// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing our Counter Component - has State of 0 with Load. Renders correctly 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
